Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs

```diff
@@ -270,17 +270,18 @@ private static string GetChannelName(string extInf, Dictionary<string, string> a
             var nameParts = extInf.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
             var nameInExtInf = nameParts.Length > 1 ? nameParts.Last().Trim() : null;
 
-            // Check for channel number with the format from SatIp
+            // Remove channel number with the format from SatIp but skip it when the tv-chno attribute is present
             // #EXTINF:0,84. VOX Schweiz
             // #EXTINF:0,84.0 - VOX Schweiz
-            if (!string.IsNullOrWhiteSpace(nameInExtInf))
+            // #EXTINF:0 tvg-chno="6" tvg-name"3+", 3+ HD
+            if (!string.IsNullOrWhiteSpace(nameInExtInf) && !attributes.ContainsKey("tvg-chno"))
             {
                 var numberIndex = nameInExtInf.IndexOf(' ');
                 if (numberIndex > 0)
                 {
                     var numberPart = nameInExtInf.Substring(0, numberIndex).Trim(new[] { ' ', '.' });
 
-                    if (double.TryParse(numberPart, NumberStyles.Any, CultureInfo.InvariantCulture, out _))
+                    if (double.TryParse(numberPart, NumberStyles.Float, CultureInfo.InvariantCulture, out _))
```

Comment:
Judging by the documented number styles and the fact that we split on commas and strip leading and trailing whitespace, the only number style left is `AllowDecimalPoint`. `NumberStyles.Float` allows exponents and leading signs, neither of which make any sense for a channel number.
```suggestion
                    if (double.TryParse(numberPart, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture, out _))
```

Reply:
Another question is, should the function `IsValidChannelNumber` be changed as well? And why isn't it used here?